test/recipes/70-test_sslcbcpadding.t

```diff
@@ -0,0 +1,107 @@
+#! /usr/bin/env perl
+# Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use OpenSSL::Test qw/:DEFAULT cmdstr srctop_file bldtop_dir/;
+use OpenSSL::Test::Utils;
+use TLSProxy::Proxy;
+
+my $test_name = "test_sslcbcpadding";
+setup($test_name);
+
+plan skip_all => "TLSProxy isn't usable on $^O"
+    if $^O =~ /^(VMS|MSWin32)$/;
+
+plan skip_all => "$test_name needs the dynamic engine feature enabled"
+    if disabled("engine") || disabled("dynamic-engine");
+
+plan skip_all => "$test_name needs the sock feature enabled"
+    if disabled("sock");
+
+plan skip_all => "$test_name needs TLSv1.2 enabled"
+    if disabled("tls1_2");
+
+$ENV{OPENSSL_ia32cap} = '~0x200000200000000';
+my $proxy = TLSProxy::Proxy->new(
+    \&add_maximal_padding_filter,
+    cmdstr(app(["openssl"]), display => 1),
+    srctop_file("apps", "server.pem"),
+    (!$ENV{HARNESS_ACTIVE} || $ENV{HARNESS_VERBOSE})
+);
+
+plan tests => 1 + 256;
+
+my $bad_padding_offset = -1;
+
+# Test 1: Maximally-padded records are accepted.
+$proxy->start() or plan skip_all => "Unable to start up Proxy for tests";
+ok(TLSProxy::Message->success(), "Maximally-padded record test");
+
+# Tests 2 through 257: Invalid padding.
+for ($bad_padding_offset = 0; $bad_padding_offset < 256;
+     $bad_padding_offset++) {
+    $proxy->clear();
+    $proxy->start() or plan skip_all => "Unable to start up Proxy for tests";;
+    ok(TLSProxy::Message->fail(), "Invalid padding byte $bad_padding_offset");
+}
+
+sub add_maximal_padding_filter
+{
+    my $proxy = shift;
+
+    if ($proxy->flight == 0) {
+        # Disable Encrypt-then-MAC.
+        foreach my $message (@{$proxy->message_list}) {
+            if ($message->mt != TLSProxy::Message::MT_CLIENT_HELLO) {
+                next;
+            }
+
+            $message->delete_extension(TLSProxy::Message::EXT_ENCRYPT_THEN_MAC);
+            $message->process_extensions();
+            $message->repack();
+        }
+    }
+
+    if ($proxy->flight == 3) {
+        # Insert a maximally-padded record. Assume a block size of 16 (AES) and
+        # a MAC length of 20 (SHA-1).
+        my $block_size = 16;
+        my $mac_len = 20;
```

Comment:
The cipher is hard-coded into the Proxy in https://github.com/openssl/openssl/blob/master/util/TLSProxy/Proxy.pm#L51 - AES128-SHA is the only one the ossltest engine can currently mock out. If you wanted to be really proper about it, I suppose you could add a cipher -> params lookup to the proxy, but I think it's fine to hard-code them here, esp since this test is explicitly about CBC.